tests/implicit/timplicit.nim

```diff
@@ -39,10 +39,10 @@ block:
   var x: ref int
   new(x)
 
-  x.p(44)
+  x[].p(44)
 
   var indirect = p
-  x.indirect(44)
+  x[].indirect(44)
 
   echo x[]
 
```

Comment:
This entire test is for implicitDeref, except the `sp.a` part. And it occupies its own folder `implicit`.

IMO move lines 18-30 to their own test (maybe `objects/tptrobjectfield.nim`), replace `if sp.a == 2: echo 2` with `doAssert sp.a == 2`, then delete this test and the folder it's in.